src/credentials_obfuscation_app.erl

```diff
@@ -25,36 +25,38 @@
 -behaviour(supervisor).
 -export([init/1]).
 
--export([enabled/0, passphrase/0, cipher/0, hash/0, iterations/0]).
+-export([enabled/0, secret/0, cipher/0, hash/0, iterations/0]).
 
 start(_StartType, _StartArgs) ->
-    _ = case enabled() of
-        true ->
-            T = ets:new(table_name(), [set, protected, named_table]),
-            ets:insert_new(T, {secret, crypto:strong_rand_bytes(128)}),
-            %% cipher/decipher attempt to crash now instead of at some awkward moment
-            check();
-        false ->
-            ok
-    end,
+    ok = case enabled() of
+             true ->
+                 Secret = get_secret(),
+                 T = ets:new(table_name(), [set, public, named_table]),
```

Comment:
Changing the `ets` table access to `public` is necessary because the secret may be updated in the future by another process than the one that owns the table.